drivers/wifi/nrf700x/zephyr/src/zephyr_disp_scan.c

```diff
@@ -130,9 +131,15 @@ void wifi_nrf_event_proc_disp_scan_res_zep(void *vif_ctx,
 
 	vif_ctx_zep = vif_ctx;
 
-	cb = (scan_result_cb_t) vif_ctx_zep->disp_scan_cb;
+	cb = (scan_result_cb_t)vif_ctx_zep->disp_scan_cb;
 
 	for (i = 0; i < scan_res->event_bss_count; i++) {
+		/* Limit the scan results to the configured maximum */
+		if ((CONFIG_NRF700X_SCAN_LIMIT >= 0) &&
+		    (vif_ctx_zep->scan_res_cnt >= CONFIG_NRF700X_SCAN_LIMIT)) {
+			break;
+		}
```

Comment:
If you can use a positive number as default (65535), then we can drop the first condition.

```suggestion
		/* Limit the scan results to the configured maximum */
		if ((CONFIG_NRF700X_SCAN_LIMIT >= 0) &&
		    (vif_ctx_zep->scan_res_cnt++ >= CONFIG_NRF700X_SCAN_LIMIT)) {
			break;
		}
```